### ISSUE TITLE
Database upgrade always runs and delays container start.

### ISSUE THREAD

**Issue by spinside** [bitnami/bitnami-docker-wordpress#140]
**Description**

Docker container always tries to upgrade the database.

**Steps to reproduce the issue:**

Run 1 Wordpress container.
Scale up an extra container, same image.
The following appears in the log:
```
wordpre INFO  Upgrading WordPress Database ...
```
This step takes between 7 to 25 minutes.

This causes the log to display the message:
```
$app.helpers.upgradeDatabase = function() {
  $app.info('Upgrading WordPress Database ...');
  const wordpressUpgradeDBScriptPath = $os.createTempFile();
  $hb.renderToFile('upgradeDB.php.tpl', wordpressUpgradeDBScriptPath);
  php.execute(wordpressUpgradeDBScriptPath, {cwd: $app.installdir});
};
```

This function is invoked by the "upgradeDB.php.tpl":

```
<?php
  require_once( '{{$app.installdir}}/wp-config.php' );
  require_once( '{{$app.installdir}}/wp-admin/includes/upgrade.php' );
  $wp_current_db_version = __get_option( 'db_version' );
  if ( $wp_db_version != $wp_current_db_version ) {
    wp_upgrade();
    exit(0);
  }
?>
```
This seems to check first if the version of the DB is the same. But the function is always executed.

**Describe the results you received:**

Database upgrade function runs every time you start a new container

**Describe the results you expected:**

If the database version is equal to the current container Wordpress version the upgrade will be skipped.

**Additional information you deem important (e.g. issue happens only occasionally):**

I'm running this on Azure Kubernetes Service. I installed everything using the helm chart stable/wordpress. Using Azure files as persistent storage and a MariaDB database 

**Version**

- Output of `docker version`:

```
Client:
 Version:      1.13.1
 API version:  1.26
 Go version:   go1.7.5
 Git commit:   092cba3
 Built:        Wed Feb  8 06:50:14 2017
 OS/Arch:      linux/amd64

Server:
 Version:      1.13.1
 API version:  1.26 (minimum version 1.12)
 Go version:   go1.7.5
 Git commit:   092cba3
 Built:        Wed Feb  8 06:50:14 2017
 OS/Arch:      linux/amd64
 Experimental: false
```

- Output of `docker info`:

```
docker info
Containers: 85
 Running: 40
 Paused: 0
 Stopped: 45
Images: 65
Server Version: 1.13.1
Storage Driver: overlay2
 Backing Filesystem: extfs
 Supports d_type: true
 Native Overlay Diff: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host macvlan null overlay
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: aa8187dbd3b7ad67d8e5e3a15115d3eef43a7ed1
runc version: 9df8b306d01f59d3a8029be411de015b7304dd8f
init version: 949e6fa
Security Options:
 apparmor
 seccomp
  Profile: default
Kernel Version: 4.15.0-1019-azure
Operating System: Ubuntu 16.04.5 LTS
OSType: linux
Architecture: x86_64
CPUs: 1
Total Memory: 3.361 GiB
Name: aks-agentpool-13583930-0
ID: SMXI:4D4M:ALIC:QFAS:YLJY:7L22:67X5:76CD:OLO2:VRCB:33IS:53BE
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
WARNING: No swap limit support
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: true
```

- Output of `docker-compose version` (if applicable):

```
-
```

**Additional environment details (AWS, VirtualBox, Docker for MAC, physical, etc.):**
Azure Kubernetes Service

Extra question, were can I find the source for the com.bitnami.wordpress? is there a Github page for this?

**Comment by spinside** [comment]
After further investigation with a higher log level it seems that upgrading the DB is not the problem.

```
wordpre INFO  Upgrading WordPress Database ...
php     TRACE [normalize] File normalization: bin -> /opt/bitnami/php/bin
php     TRACE [normalize] File normalization: bin -> /opt/bitnami/php/bin
php     TRACE [runProgram] Executing: /opt/bitnami/php/bin/php /tmp/15337209119260.29217231359824947
php     TRACE [runProgram] Executing internal command: '/bin/sh' ["-c","/opt/bitnami/php/bin/php /tmp/15337209119260.29217231359824947"]
php     TRACE [runProgram] RESULT: {"code":0,"stderr":"PHP Notice:  Undefined index: HTTP_HOST in /bitnami/wordpress/wp-config.php on line 86\nPHP Notice:  Undefined index: HTTP_HOST in /bitnami/wordpress/wp-config.php on line 87\n","stdout":""}
wordpre TRACE [normalize] File normalization: tmp -> /opt/bitnami/wordpress/tmp
wordpre TRACE [isLink] File normalization: /opt/bitnami/wordpress -> /opt/bitnami/wordpress
wordpre TRACE [isDirectory] File normalization: /opt/bitnami/wordpress -> /opt/bitnami/wordpress
wordpre TRACE [configurePermissions] File to chown: /opt/bitnami/wordpress
wordpre TRACE [chown] File normalization: /opt/bitnami/wordpress -> /opt/bitnami/wordpress
```
The chown proces takes a long time. This should be run-once as said in: https://github.com/bitnami/bitnami-docker-wordpress/issues/136

So the extra question still stands: Where can I find the source for the com.bitnami.wordpress? is there a Github page for this? I would like to fix this issue by adding a check if the permissions are already set correctly.

**Comment by javsalgar** [comment]
I saw that you just responded when I was going to share my findings :D. Indeed, the database was not cause of the performance issue (I also instrumented the php code to make sure that it was taking the correct database version). 

The thing with the chmod/chowns is how to identify that it was done before by another container. We could implement some kind of logic where each node has an id and have only one container perform the operation. I also have my concerns about how to fix a permission issue in the volume if it is only run once. As @tompizmor said, that is something we will need to study, but we cannot provide an ETA due to the issue not being critical. However, stay tuned!
